### PR TITLE
Changing logging from log4j1.2 to logback + adding logging to file

### PR DESCRIPTION
**Pull request by MansuyDavid**
Hello,

At the begining I wanted to add a file appender to the loging.
But when I tried to do that I realized that even if the POM had a dependency of log4j-core 2.22.1, this one is never used.
The slf4j-log4j12 depend on "slf4j-reload4j" which depend on "reload4j" which is "The reload4j project is a fork of Apache log4j version 1.2.17 in order to fix most pressing security issues".
That's why the log4j.properties, is not a log4j2.properties, and the property declaration isn't done in the same way between 1.2 and 2.x.

So I wanted to update log4j1.2 (because log4j1.2 is quite old, and this project is very young and don't use old  fird party libraries with log4j, so why not using a more mainstream log system?)

At first I switch to log4j2 in order to not change completely what you had done.
But during my tests on native image, I found that log4j2 doesn't support native compilation for now. Most github comments says that it is recommanded to switch to logback.
Which I did, and all worked perfectly.

Hope you'll find that usefull.
David.

### CODE REVIEW

**Comment by loyispa** [comment]
@MansuyDavid  Thanks for your great works.  I agree with your thoughts for log4j, i left some comments, please take a look.

**Review by loyispa — CHANGES_REQUESTED** [review]

File: src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%d{YYYY-MM-dd HH:mm:ss.SSS} [%thread] %-5level %logger{36}%L -%kvp- %msg%n</pattern>
```
[the remark this answers is not in the document]
Comment (MansuyDavid) [reply src/main/resources/logback.xml]: indeed! I corrected this

File: src/main/resources/logback.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        </encoder>
+    </appender>
+
+    <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>./logs/jgc_exporter.log</file>
```
[the remark this answers is not in the document]
Comment (MansuyDavid) [reply src/main/resources/logback.xml:10]: absolutely, I've changed the gitignore

**Comment by loyispa** [comment]
Thank you.

**Pull request merged**